src/utils/breakpoint/tests/astBreakpointLocation.spec.js

```diff
@@ -0,0 +1,34 @@
+import { getASTLocation } from "../astBreakpointLocation.js";
+import { getSource } from "../../parser/tests/helpers";
+
+describe("ast", () => {
+  describe("valid location", () => {
+    it("returns the scope and offset", async () => {
+      const source = getSource("math");
+      // mock out immutable.js
+      source.toJS = () => source;
+
+      const location = { line: 6, column: 0 };
+      const astLocation = await getASTLocation(source, location);
+      const offset = astLocation.scope.location.start;
+      expect(astLocation.scope.name).toBe("math");
+      expect(astLocation.offset.line).toBe(location.line - offset.line);
+      expect(astLocation.offset.column).toBe(location.column);
+      expect(astLocation).toMatchSnapshot();
+    });
+  });
```

Comment:
lets add a test for nested functions and just lock down a behavior. I think we want to use the closest named function